data-prepper-core/src/main/java/com/amazon/dataprepper/parser/config/MetricsConfig.java

```diff
@@ -60,21 +81,78 @@ public JvmThreadMetrics jvmThreadMetrics() {
         return new JvmThreadMetrics();
     }
 
+    private void configureMetricRegistry(final MeterRegistry meterRegistry) {
+        meterRegistry.config()
+                .commonTags(Collections.singletonList(
+                        Tag.of(SERVICE_NAME, getServiceNameForMetrics())
+                ));
+
+    }
+
+    @Bean
+    public Optional<MeterRegistry> prometheusMeterRegistry(
+            final DataPrepperConfiguration dataPrepperConfiguration,
+            final Optional<Authenticator> optionalAuthenticator,
```

Comment:
Taking an `Optional` as an argument is not a good practice. You have no non-null guarantees here, so you should do a null check, which somewhat defeats the purpose.

I noticed this in a few places. Please change these to just take the argument in without the Optional wrapper.

Reply:
Depending on your Data Prepper configuration and Pipeline an Authenticator Bean may not be created / available. By default, if a requested bean is not available Spring will abort startup throwing an exception. We have three options in how to handle a case like this:

1. Always create an Authenticator, then use it based on configuration.
2. Wrapping the dependency as an Optional - JSR-330 compliant and would require the lease amount of refactoring if we switch to another DI framework like Dagger
3. Use the Spring native functionality @Autowired(required = false)

Another good example to consider is when creating the CloudWatchMeterRegistry. If the CloudWatchMeterRegistry is not configured, creating an instance of it cloud cause an exception to be thrown,

https://github.com/sbayer55/data-prepper/blob/di-server/data-prepper-core/src/main/java/com/amazon/dataprepper/parser/config/MetricsConfig.java#L120

Reply:
@sbayer55, why not go with option 1? That sounds like it decouples DP with a dependency framework and won't require passing optionals around.

Reply:
Option 1 presents issues with classes like CloudWatchMeterRegistry that would need additional refactoring. Option 3 is a happy middle ground. We remove the need for optionals while limiting the scope of this PR.